editor/config_loader.hpp

```diff
@@ -0,0 +1,77 @@
+#pragma once
+
+#include "base/exception.hpp"
+#include "base/logging.hpp"
+
+#include "std/chrono.hpp"
+#include "std/mutex.hpp"
+#include "std/string.hpp"
+#include "std/thread.hpp"
+
+namespace pugi
+{
+class xml_document;
+}
+
+namespace editor
+{
+// Class for multithreaded interruptable waiting.
+class Waiter
+{
+public:
+  template <typename Rep, typename Period>
+  bool Wait(duration<Rep, Period> const & d)
+  {
+    unique_lock<mutex> lock(m_mutex);
+
+    if (m_interrupted)
+      return false;
+
+    try
+    {
+      m_event.wait_for(lock, d, [this]() { return m_interrupted; });
+    }
+    catch (RootException const & ex)
```

Comment:
Could you please explain what part of m_event.wait_for may throw a RootException?